### PR TITLE
Feature: Add support for localstack

### PR DESCRIPTION
**Pull request by joanmiespada**
This change adds support for localstack image, latest version.

localstack: https://hub.docker.com/r/localstack/localstack

### CODE REVIEW

File: testcontainers/src/images/local_stack.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    fn name(&self) -> String {
+        NAME.to_owned()
+    }
+
+    fn tag(&self) -> String {
+        self.tag.to_owned()
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![WaitFor::message_on_stdout( format!("Starting edge router (https port {})...", PORT))]
```
Comment (mdelapenya) [inline testcontainers/src/images/local_stack.rs:58]: Hi from testcontainers-go 👋 

I'm jumping here to share that in other languages we define a different waitFor log strategy:
- Go: https://github.com/testcontainers/testcontainers-go/pull/800/files#diff-bd71ec18d411b5b0b51723761d6d0a95e60ad7d5578828df544e14d6518a222dR53
- Java: https://github.com/testcontainers/testcontainers-java/blob/343c5a533f4c6148d6e6eea3a3374116365bc91d/modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java#L98

We can probably ping @eddumelendez and @kiview for creating alignment across different implementations, but I'm not against setting this one as the default, so it's not a blocker for me if the maintainers decide to merge it :)

**Comment by joanmiespada** [comment]
Hi @mdelapenya, thanks for your comment. I've updated it, in the same way as go and java. Please, check.

File: testcontainers/src/images/local_stack.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    volumes: HashMap<String, String>,
+    tag: String,
+}
+
+impl LocalStack{
+    pub fn set_services(&mut self, services: &str) {
+
+        self.env_vars.insert("SERVICES".to_owned(), services.to_owned());
+
+    }
```
Comment (HofmeisterAn) [inline testcontainers/src/images/local_stack.rs:20]: Having an API to configure the services seems unnecessary for the latest versions of LocalStack. The services are "available" immediately anyway.

For Testcontainers implementations that do not need to support backwards compatibility (older versions) like .NET and Go we decided not to add an API to configure the services. Developers can simply use the API that configures environment variables (if necessary).

File: testcontainers/src/images/local_stack.rs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    fn default() -> Self {
+        let mut env_vars = HashMap::new();
+        //env_vars.insert("SERVICES".to_owned(), "s3".to_owned());
+        env_vars.insert("DEBUG".to_owned(), "1".to_owned());
+        env_vars.insert("PORT_WEB_UI".to_owned(), "8080".to_owned());
+        env_vars.insert("LAMBDA_EXECUTOR".to_owned(), "docker".to_owned());
+        env_vars.insert("DOCKER_HOST".to_owned(), "unix:///var/run/docker.sock".to_owned());
+        env_vars.insert("AWS_DEFAULT_REGION".to_owned(), "eu-central-1".to_owned());
+        env_vars.insert("AWS_ACCESS_KEY_ID".to_owned(), "test".to_owned());
+        env_vars.insert("AWS_SECRET_ACCESS_KEY".to_owned(), "test".to_owned());
```
Comment (HofmeisterAn) [inline testcontainers/src/images/local_stack.rs:33]: I do not think these environment variables are necessary.

File: testcontainers/src/images/local_stack.rs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    fn name(&self) -> String {
+        NAME.to_owned()
+    }
+
+    fn tag(&self) -> String {
+        self.tag.to_owned()
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![WaitFor::message_on_stdout( format!("Ready."))]
```
Comment (HofmeisterAn) [inline testcontainers/src/images/local_stack.rs:58]: Recent versions should use the health check endpoint `/_localstack/health` to detect readiness of the container.

File: testcontainers/src/images/local_stack.rs [hunk 5]
```diff
@@ -0,0 +1,72 @@
+use crate::{core::WaitFor, Image};
+use std::collections::HashMap;
+
+const NAME: &str = "localstack/localstack";
+const TAG: &str = "latest";
```
Comment (HofmeisterAn) [inline testcontainers/src/images/local_stack.rs:5]:
```suggestion
const TAG: &str = "1.4";
```

Pin the version, maybe even to `1.4.0`.

**Review by HofmeisterAn — COMMENTED** [review]
We had some discussions today about the LocalStack implementation in .NET and Go. I would like to share our thoughts (see comments below).

File: testcontainers/src/images/local_stack.rs (hunk 5 above)
Comment (joanmiespada) [reply testcontainers/src/images/local_stack.rs:5]: agree

File: testcontainers/src/images/local_stack.rs (hunk 2 above)
Comment (joanmiespada) [reply testcontainers/src/images/local_stack.rs:20]: ok

File: testcontainers/src/images/local_stack.rs (hunk 3 above)
Comment (joanmiespada) [reply testcontainers/src/images/local_stack.rs:33]: let's remove them.

File: testcontainers/src/images/local_stack.rs (hunk 4 above)
Comment (joanmiespada) [reply testcontainers/src/images/local_stack.rs:58]: ok, but it'd requiere more effort parsing the /health response, right? or do you suggest to check it it reponses with HTTP 200/OK?

Comment (HofmeisterAn) [reply testcontainers/src/images/local_stack.rs:46]: Yes, we check for HTTP 200/OK.

Comment (HofmeisterAn) [reply testcontainers/src/images/local_stack.rs:46]: I'm afraid my knowledge of the Rust implementation is limited. Are you saying there is no wait strategy available to wait for an HTTP status code? Perhaps we could use `WaitFor::Healthcheck` or add the strategy to Rust as a new feature. That would be really useful.

**Comment by thomaseizinger** [comment]
We will be removing all images from this library, see https://github.com/testcontainers/testcontainers-rs/issues/471.

**Pull request closed without merging**